src/hooks/selector.ts

```diff
@@ -0,0 +1,97 @@
+import { useEffect } from 'react';
+
+import { SelectorConfig } from '@isrd-isi-edu/deriva-webapps/src/models/plot';
+import { PlotTemplateParams } from '@isrd-isi-edu/deriva-webapps/src/hooks/chart';
+import { getQueryParam } from '@isrd-isi-edu/chaise/src/utils/uri-utils';
+import { windowRef } from '@isrd-isi-edu/chaise/src/utils/window-ref';
+import { Option } from '@isrd-isi-edu/deriva-webapps/src/components/virtualized-select';
+
+type SelectorsGridProps = {
+    /**
+     * selectors data to be rendered
+     */
+    selectorConfig: SelectorConfig[];
+    templateParams: PlotTemplateParams;
+    setDataOptions: any;
+};
+
+/**
+ * This function retrieves dropdown options from its configuration, 
+ * mapping them into an array of label-value pairs, and returns an array of arrays containing these options.
+ * @param selectorGridObject Selector configuration, template params and setDataOptions state method
+ * @returns array of arrays containing these options
+ */
+const getDataOptions = async (selectorGridObject: SelectorsGridProps) => {
+    //TODO: To fetch selector options from url_pattern
+    // if (selectorGridObject?.selectorConfig?.request_info?.url_pattern) {
+    //     const pattern = selectorGridObject?.selectorConfig?.request_info?.url_pattern;
+    //     const { uri, headers } = getPatternUri(pattern, selectorGridObject?.templateParams);
+    //     const { initialReference, tupleData } = await fetchSelectGridCellData(
+    //         uri,
+    //         headers
+    //     ); // perform the data fetch
+    //     return { initialReference, tupleData };
+    // } else {
+        const allSelectorDataOptions: Option[][]=[];
+        selectorGridObject?.selectorConfig?.map((currentConfig)=>{
+            const dataOption = currentConfig?.request_info?.data;
+            const dropdownOptions: { label: any; value: any; }[] = [];
+            dataOption?.map((option: any) => {
+                dropdownOptions.push({ 'label': option.Display, 'value': option.Name });
+            });
+            allSelectorDataOptions.push(dropdownOptions);
+        })
+        
+        return allSelectorDataOptions;
+    // }
+}
+
+/**
+ * It iterates through its selectorConfig for multiple selectors. 
+ * For each configuration, it extracts values for the selector and sets them in $control_values object within configData.templateParams under the selector's uid
+ * and then returns the modified configData.templateParams.
+ * @param configData Selector configuration, template params and setDataOptions state method
+ * @returns modified configData.templateParams
+ */
+const changeSelectorData = (configData: SelectorsGridProps) => {
+    configData.selectorConfig.map((currentConfig: SelectorConfig)=>{
```

Comment:
Throws an error if one plot has a `selectorConfig` defined and the other does not. There should be a check done here or before calling this function in the userEffect